src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -57,5 +69,288 @@ private void Initialize(IEdmEntityType entityType)
+                    }
+                }
+                else
+                {
+                    //call patchitem if its single structuredobj
+                    EdmStructuredObject structuredObj = value as EdmStructuredObject;
+
+                    if (structuredObj != null)
+                    {
+                        object obj;
+                        originalObj.TryGetPropertyValue(propertyName, out obj);
+
+                        EdmStructuredObject origStructuredObj = obj as EdmStructuredObject;
+
+                        if(origStructuredObj == null)
+                        {
+                            if(structuredObj is EdmComplexObject)
+                            {
+                                origStructuredObj = new EdmComplexObject(structuredObj.ActualEdmType as IEdmComplexType);
+                            }
+                            else
+                            {
+                                origStructuredObj = new EdmEntityObject(structuredObj.ActualEdmType as IEdmEntityType);
+                            }
+
+                            originalObj.TrySetPropertyValue(propertyName, origStructuredObj);
+                        }
+
+                        PatchItem(structuredObj, origStructuredObj);                        
+                    }
+                    else
+                    {
+                        originalObj.TrySetPropertyValue(propertyName, value);
+                    }
+                }
+            }
+        }
+
+        private IEdmChangedObject HandleFailedOperation(EdmEntityObject changedObj, DataModificationOperationKind operation, EdmStructuredObject originalObj, 
+            List<IEdmStructuralProperty> keys, string errorMessage)
+        {
+            IEdmChangedObject edmChangedObject = null;
+            DataModificationExceptionType dataModificationExceptionType = new DataModificationExceptionType(operation);
+            dataModificationExceptionType.MessageType = new MessageType();
+            dataModificationExceptionType.MessageType.Message = errorMessage;
+
+            // This handles the Data Modification exception. This adds Core.DataModificationException annotation and also copy other instance annotations.
+            //The failed operation will be based on the protocol
+            switch (operation)
+            {
+                case DataModificationOperationKind.Update:
+                    edmChangedObject = changedObj as IEdmChangedObject;
+                    break;
+                case DataModificationOperationKind.Insert:
+                    {
+                        EdmDeltaDeletedEntityObject edmDeletedObject = new EdmDeltaDeletedEntityObject(EntityType);
+                        PatchItem(edmDeletedObject, changedObj);
+
+                        TryGetContentId(changedObj, keys, edmDeletedObject);
+
+                        edmDeletedObject.TransientInstanceAnnotationContainer = changedObj.TransientInstanceAnnotationContainer;
```

Comment:
should we be using the transient instance annotation container from changedObj, or creating a new transient instance annotation container for the response?  Will we have copied any transientInstanceAnnotations to changedObj?

Reply:
Any transient annotations should carry, right